### ISSUE TITLE
ERROR: Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1)

### ISSUE THREAD

**Issue by BramvdHoek** [haugene/docker-transmission-openvpn#1286]
**Describe the problem**
Docker container did run fine, reinstalled my raspberry pi today and when I setup everything again (didn't make a backup), I get the following error:
ERROR: Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1)

**Add your docker run command**
I did setup everything in portainer, PUID and PGID are from root:

> root@DietPi:~# id root
> uid=0(root) gid=0(root) groups=0(root)

```
OPENVPN_USERNAME | <iscorrect>
OPENVPN_PASSWORD | <iscorrect>
OPENVPN_PROVIDER | PIA
TRANSMISSION_WEB_UI | transmission-web-control
WEBPROXY_ENABLED | true
PUID | 0
PGID | 0
LOCAL_NETWORK | 192.168.1.0/24
OPENVPN_OPTS | --inactive 3600 --ping 10 --ping-exit 60
PATH | /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
LC_ALL | C.UTF-8
DEBIAN_FRONTEND | noninteractive
UDEV | off
QEMU_CPU | arm1176
GLOBAL_APPLY_PERMISSIONS | true
TRANSMISSION_ALT_SPEED_DOWN | 50
TRANSMISSION_ALT_SPEED_ENABLED | false
TRANSMISSION_ALT_SPEED_TIME_BEGIN | 540
TRANSMISSION_ALT_SPEED_TIME_DAY | 127
TRANSMISSION_ALT_SPEED_TIME_ENABLED | false
TRANSMISSION_ALT_SPEED_TIME_END | 1020
TRANSMISSION_ALT_SPEED_UP | 50
TRANSMISSION_BIND_ADDRESS_IPV4 | 0.0.0.0
TRANSMISSION_BIND_ADDRESS_IPV6 | ::
TRANSMISSION_BLOCKLIST_ENABLED | false
TRANSMISSION_BLOCKLIST_URL | http://www.example.com/blocklist
TRANSMISSION_CACHE_SIZE_MB | 4
TRANSMISSION_DHT_ENABLED | true
TRANSMISSION_DOWNLOAD_DIR | /data/completed
TRANSMISSION_DOWNLOAD_LIMIT | 100
TRANSMISSION_DOWNLOAD_LIMIT_ENABLED | 0
TRANSMISSION_DOWNLOAD_QUEUE_ENABLED | true
TRANSMISSION_DOWNLOAD_QUEUE_SIZE | 5
TRANSMISSION_ENCRYPTION | 1
TRANSMISSION_IDLE_SEEDING_LIMIT | 30
TRANSMISSION_IDLE_SEEDING_LIMIT_ENABLED | false
TRANSMISSION_INCOMPLETE_DIR | /data/incomplete
TRANSMISSION_INCOMPLETE_DIR_ENABLED | true
TRANSMISSION_LPD_ENABLED | false
TRANSMISSION_MAX_PEERS_GLOBAL | 200
TRANSMISSION_MESSAGE_LEVEL | 2
TRANSMISSION_PEER_ID_TTL_HOURS | 6
TRANSMISSION_PEER_LIMIT_GLOBAL | 200
TRANSMISSION_PEER_LIMIT_PER_TORRENT | 50
TRANSMISSION_PEER_PORT | 51413
TRANSMISSION_PEER_PORT_RANDOM_HIGH | 65535
TRANSMISSION_PEER_PORT_RANDOM_LOW | 49152
TRANSMISSION_PEER_PORT_RANDOM_ON_START | false
TRANSMISSION_PEER_SOCKET_TOS | default
TRANSMISSION_PEX_ENABLED | true
TRANSMISSION_PORT_FORWARDING_ENABLED | false
TRANSMISSION_PREALLOCATION | 1
TRANSMISSION_PREFETCH_ENABLED | 1
TRANSMISSION_QUEUE_STALLED_ENABLED | true
TRANSMISSION_QUEUE_STALLED_MINUTES | 30
TRANSMISSION_RATIO_LIMIT | 2
TRANSMISSION_RATIO_LIMIT_ENABLED | false
TRANSMISSION_RENAME_PARTIAL_FILES | true
TRANSMISSION_RPC_AUTHENTICATION_REQUIRED | false
TRANSMISSION_RPC_BIND_ADDRESS | 0.0.0.0
TRANSMISSION_RPC_ENABLED | true
TRANSMISSION_RPC_HOST_WHITELIST_ENABLED | false
TRANSMISSION_RPC_PASSWORD | password
TRANSMISSION_RPC_PORT | 9091
TRANSMISSION_RPC_URL | /transmission/
TRANSMISSION_RPC_USERNAME | username
TRANSMISSION_RPC_WHITELIST | 127.0.0.1
TRANSMISSION_RPC_WHITELIST_ENABLED | false
TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED | true
TRANSMISSION_SCRIPT_TORRENT_DONE_ENABLED | false
TRANSMISSION_SEED_QUEUE_ENABLED | false
TRANSMISSION_SEED_QUEUE_SIZE | 10
TRANSMISSION_SPEED_LIMIT_DOWN | 100
TRANSMISSION_SPEED_LIMIT_DOWN_ENABLED | false
TRANSMISSION_SPEED_LIMIT_UP | 100
TRANSMISSION_SPEED_LIMIT_UP_ENABLED | false
TRANSMISSION_START_ADDED_TORRENTS | true
TRANSMISSION_TRASH_ORIGINAL_TORRENT_FILES | false
TRANSMISSION_UMASK | 2
TRANSMISSION_UPLOAD_LIMIT | 100
TRANSMISSION_UPLOAD_LIMIT_ENABLED | 0
TRANSMISSION_UPLOAD_SLOTS_PER_TORRENT | 14
TRANSMISSION_UTP_ENABLED | false
TRANSMISSION_WATCH_DIR | /data/watch
TRANSMISSION_WATCH_DIR_ENABLED | true
TRANSMISSION_HOME | /data/transmission-home
TRANSMISSION_WATCH_DIR_FORCE_GENERIC | false
ENABLE_UFW | false
UFW_ALLOW_GW_NET | false
UFW_DISABLE_IPTABLES_REJECT | false
WEBPROXY_PORT | 8888
DOCKER_LOG | false
HEALTH_CHECK_HOST | google.com
OPENVPN_CONFIG | CA Toronto
CREATE_TUN_DEVICE | true
TRANSMISSION_PEER_CONGESTION_ALGORITHM |  
TRANSMISSION_RPC_HOST_WHITELIST |  
TRANSMISSION_SCRIPT_TORRENT_DONE_FILENAME |  
UFW_EXTRA_PORTS |  
TRANSMISSION_WEB_HOME |  
DROP_DEFAULT_ROUTE
```

**Logs**
```
Using OpenVPN provider: PIA,
Starting OpenVPN using config CA Toronto.ovpn,
Setting OPENVPN credentials...,
adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0,
RTNETLINK answers: Operation not permitted,
Wed Jul  8 21:25:56 2020 OpenVPN 2.4.7 arm-unknown-linux-gnueabihf [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Feb 20 2019,
Wed Jul  8 21:25:56 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10,
Wed Jul  8 21:25:56 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts,
Wed Jul  8 21:25:56 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]172.98.67.2:1198,
Wed Jul  8 21:25:56 2020 UDP link local: (not bound),
Wed Jul  8 21:25:56 2020 UDP link remote: [AF_INET]172.98.67.2:1198,
Wed Jul  8 21:25:56 2020 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this,
Wed Jul  8 21:25:56 2020 [603b4b0fee1435a3662ac1ae126f7c6b] Peer Connection Initiated with [AF_INET]172.98.67.2:1198,
Wed Jul  8 21:25:58 2020 ERROR: Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1),
Wed Jul  8 21:25:58 2020 Exiting due to fatal error,
Using OpenVPN provider: PIA,
Starting OpenVPN using config CA Toronto.ovpn,
Setting OPENVPN credentials...,
adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0,
RTNETLINK answers: Operation not permitted,
Wed Jul  8 21:26:01 2020 OpenVPN 2.4.7 arm-unknown-linux-gnueabihf [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Feb 20 2019,
Wed Jul  8 21:26:01 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10,
Wed Jul  8 21:26:01 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts,
Wed Jul  8 21:26:01 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]172.98.67.93:1198,
Wed Jul  8 21:26:01 2020 UDP link local: (not bound),
Wed Jul  8 21:26:01 2020 UDP link remote: [AF_INET]172.98.67.93:1198,
Wed Jul  8 21:26:01 2020 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this,
Wed Jul  8 21:26:01 2020 [e1438ae6d6938ff8fbb7b6aa72ba5f87] Peer Connection Initiated with [AF_INET]172.98.67.93:1198,
Wed Jul  8 21:26:03 2020 ERROR: Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1),
Wed Jul  8 21:26:03 2020 Exiting due to fatal error,
Using OpenVPN provider: PIA,
Starting OpenVPN using config CA Toronto.ovpn,
Setting OPENVPN credentials...,
adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0,
RTNETLINK answers: Operation not permitted,
Wed Jul  8 21:26:06 2020 OpenVPN 2.4.7 arm-unknown-linux-gnueabihf [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Feb 20 2019,
Wed Jul  8 21:26:06 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10,
Wed Jul  8 21:26:06 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts,
Wed Jul  8 21:26:06 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]172.98.67.2:1198,
Wed Jul  8 21:26:06 2020 UDP link local: (not bound),
Wed Jul  8 21:26:06 2020 UDP link remote: [AF_INET]172.98.67.2:1198,
Wed Jul  8 21:26:06 2020 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this,
Wed Jul  8 21:26:07 2020 [603b4b0fee1435a3662ac1ae126f7c6b] Peer Connection Initiated with [AF_INET]172.98.67.2:1198,
Wed Jul  8 21:26:08 2020 ERROR: Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1),
Wed Jul  8 21:26:08 2020 Exiting due to fatal error,
```
**This error x100 times **

**Host system:**

Raspberry Pi 4 (4GB) running Dietpi:

> [  OK  ] DietPi-Update | No update required, your DietPi installation is already up to date:
> [ INFO ] DietPi-Update | Current version : v6.31.2
> [ INFO ] DietPi-Update | Latest version  : v6.31.2

```
Docker:
Client: Docker Engine - Community
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a6621
 Built:             Mon Jun 22 15:53:41 2020
 OS/Arch:           linux/arm
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a6621
  Built:            Mon Jun 22 15:47:34 2020
  OS/Arch:          linux/arm
  Experimental:     false
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

```

**Comment by BramvdHoek** [comment]
OPENVPN username and password are correct, did leave them blank here for privacy matters

**Comment by pkishino** [comment]
well, there is some permission issue as you can see.
https://www.linuxquestions.org/questions/linux-networking-3/openvpn-cannot-ioctl-tunsetiff-tun-operation-not-permitted-571942/

**Comment by BramvdHoek** [comment]
> well, there is some permission issue as you can see.
> https://www.linuxquestions.org/questions/linux-networking-3/openvpn-cannot-ioctl-tunsetiff-tun-operation-not-permitted-571942/
> try and create a tun device manually and then use that

Tried the solutions that where recommended when I did google the following error line in my log:
`Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1)`
Unfortenatly It didn't work.

But I found the problem: it is this line in my error log:
`RTNETLINK answers: Operation not permitted,`

How did I solve it? 
`--cap-add=NET_ADMIN -d ` didn't work in the enviroment fields (used portainer). I'm running it on a raspberry pi 4, so maybe there is another required syntax. However on portainer I can click on the **capabilities** tab under **advanced container settings**, there I can toggle the switch **NET_ADMIN**. Deploying the container works.

So if anyone else encounters this problem on a Raspberry PI, this might be a little workaround!
